packages/components/src/InputTime/InputTime.rebuilt.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { ChangeEvent, useId, useRef } from "react";
+import { useTimePredict } from "./hooks/useTimePredict";
+import { InputTimeProps, InputTimeRebuiltProps } from "./InputTime.Types";
+import { dateToTimeString, timeStringToDate } from "./utils/input-time-utils";
+import { FormFieldWrapper, useFormFieldWrapperStyles } from "../FormField";
+
+export function InputTimeRebuilt({
+  defaultValue,
+  value,
+  onChange,
+  ...params
+}: InputTimeRebuiltProps) {
+  const ref =
+    (params.inputRef as React.RefObject<HTMLInputElement>) ??
+    useRef<HTMLInputElement>(null);
+  const { setTypedTime } = useTimePredict({
+    value,
+    handleChange,
+  });
+
+  const { inputStyle } = useFormFieldWrapperStyles(params);
+
+  const id = getId(params);
+
+  return (
+    <FormFieldWrapper
+      disabled={params.disabled}
+      size={params.size}
+      align={params.align}
+      inline={params.inline}
+      name={params.name}
+      error={""}
+      identifier={id}
+      descriptionIdentifier={`descriptionUUID--${id}`}
+      invalid={Boolean(params.invalid)}
+      description={params.description}
+      clearable={params.clearable ?? "never"}
+      onClear={handleClear}
+      placeholder={params.placeholder}
+      value={value}
+    >
+      <input
+        ref={ref}
+        type="time"
+        className={inputStyle}
+        onBlur={handleBlur}
+        onChange={handleChangeEvent}
+        onFocus={handleFocus}
+        defaultValue={defaultValue ? dateToTimeString(defaultValue) : undefined}
+        onKeyUp={e => {
+          !isNaN(parseInt(e.key, 10)) && setTypedTime(prev => prev + e.key);
+        }}
+        value={dateToTimeString(value)}
+      />
+    </FormFieldWrapper>
+  );
+
+  function handleChangeEvent(event: ChangeEvent<HTMLInputElement>) {
+    handleChange(event.target.value);
+  }
+
+  function handleChange(newValue: string) {
+    onChange?.(timeStringToDate(newValue));
+  }
```

Comment:
this is a bit silly, but the reason is that `useTimePredict` wants a handler that takes in a string as the arg, so I'd like to have a function like that.

then, for the `input` the `onChange` will provide a ChangeEvent, so we need a function for that too.

it's also an option to modify `useTimePredict` to accept the event, though I'm trying to minimize changes during this work and the other version is all about strings.